tests/test_controller.py

```diff
@@ -621,16 +777,21 @@ def test_metadata_needed_for(self):
             metadata_already_id,
         ])
 
-        with self.app.test_request_context(headers=self.valid_auth):
+        with self.authenticated_request():
             response = self.controller.metadata_needed_for(self.collection.name)
 
-        m = messages = self.get_messages(response.get_data())
+        [m] = self.get_messages(response.get_data())
```

Comment:
The old code wasn't checking that there is only a single message in the response.